selectel/mks.go

```diff
@@ -199,10 +199,61 @@ func mksClusterV1GetLatestPatchVersions(ctx context.Context, client *v1.ServiceC
 	return result, nil
 }
 
+func mksClusterV1GetLatestMinorVersion(ctx context.Context, client *v1.ServiceClient) (string, error) {
+	kubeVersions, _, err := kubeversion.List(ctx, client)
+	if err != nil {
+		return "", err
+	}
+
+	versions := map[string]struct{}{}
+	for _, version := range kubeVersions {
+		ver, err := kubeVersionTrimToMinor(version.Version)
+		if err != nil {
+			return "", fmt.Errorf("can't get major version")
+		}
+		versions[ver] = struct{}{}
+	}
+
+	// Find max supported version.
+	latestSupported := "1.0"
+	for version := range versions {
+		latestSupported, err = compareTwoKubeVersionsByMinor(latestSupported, version)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return latestSupported, nil
+}
+
+// checkVersionIsSupported check that desired k8s version is supported.
+func checkVersionIsSupported(ctx context.Context, client *v1.ServiceClient, desiredMinorVersion string) (bool, error) {
+	kubeVersions, _, err := kubeversion.List(ctx, client)
+	if err != nil {
+		return false, err
+	}
```

Comment:
It will be better if we will get `kubeVersions` list one time before and then pass this list to the function.
Also this way helps to add more simple unit tests.
For example, check `parseMKSKubeVersionsV1Latest` implementation.